cranelift/filetests/filetests/wasm/duplicate-loads-dynamic-memory.wat

```diff
@@ -4,7 +4,8 @@
 ;;!
 ;;! settings = [
 ;;!   "enable_heap_access_spectre_mitigation=true",
-;;!   "opt_level=speed_and_size"
+;;!   "opt_level=speed_and_size",
+;;!   "use_egraphs=false"
```

Comment:
For this test, and the other in this commit, I think we do want to enable egraphs (or at least use the default setting). The point of these tests is to make sure that we deduplicate the loads by default, not to exercise a particular pass.

Reply:
Indeed; this test and the corresponding static case were actually duplicated in #5594 to `-egraph.wat` copies of the files, with egraphs enabled. The point here is to keep coverage for both cases, until we delete the non-egraphs case.